include/planar_sssp/HenzingerPriorityQueue.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <networkit/graph/Graph.hpp>
+#include <set>
+
+static const int INF = std::numeric_limits<int>::max();
+
+namespace Koala {
+class HPriorityQueue {
+    std::set<std::pair<NetworKit::count, NetworKit::node>> set;
+    std::unordered_map<NetworKit::node, NetworKit::count> idMap;
+
+    // caching the mininmal elemnet to ensure constan time read
+    std::pair<NetworKit::count, NetworKit::node> minElement;
+
+   public:
+    void initialize(std::vector<NetworKit::node>& ids);
+    void insert(NetworKit::node id, NetworKit::count key);
+    void updateKey(NetworKit::node id, NetworKit::count newKey);
+    void deactivateItem(NetworKit::node id);
+    NetworKit::count minKey();
+    NetworKit::node minItem();
+    bool empty();
```

Comment:
Just for more consistency with other heap-like structures in the library:
```
    explicit HenzingerPriorityQueue(std::vector<NetworKit::node> &indices);

    const NetworKit::node& minimum_item() const;
    const value_type& minimum_key() const;
    iterator push(id, const value_type&);
    void update(id, const value_type&);
    void deactivate(id);
    void pop();
    void empty();
```
etc.